rmw_fastrtps_shared_cpp/src/rmw_get_topic_info.cpp

```diff
@@ -143,7 +144,8 @@ _set_rmw_topic_info(
     return ret;
   }
   // set topic type
-  std::string type_name = _demangle_if_ros_type(topic_data.topic_type);
+  std::string type_name =
+    no_mangle ? _demangle_if_ros_type(topic_data.topic_type) : topic_data.topic_type;
```

Comment:
Sorry, the logic is the opposite, though I have to recognize it's confusing.

no_mangle:
- When true, it means that the passed topic name will not be mangle with ros conventions, and it's directly the DDS topic name.
  In that case, we don't want to demangle the type (i.e.: transform it from DDS type name to the ros type name), we just want to see the raw DDS type name.
- When false, the the ros topic name is mangled into the DDS name. In that case, we want to see the demangled type name, and not the raw dds type name.

That's way a say updating the docs in the rmw PR is needed :rofl:. It's super confusing.

Reply:
Yes, it's definitely confusing :laughing:

> it means that the passed topic name will not be mangle with ros conventions, and it's directly the DDS topic name

Mangling means DDS topic name -> ROS topic name?

> we don't want to demangle the type (i.e.: transform it from DDS type name to the ros type name), we just want to see the raw DDS type name

Demangling means DDS type name -> ROS type name?

> the the ros topic name is mangled into the DDS name

Mangling means ROS topic name -> DDS topic name? Did you mean to say demangle here?

> the demangled type name, and not the raw dds type name

Demangling means DDS type name -> ROS type name?

Reply:
Also, the current docs just use the term "fully qualified name". I assume that means the DDS topic name, correct? Does it also apply to the DDS type name?

Reply:
> Mangling means DDS topic name -> ROS topic name? Did you mean to say "will not demangle" here?

Mangling means: ROS name -> DDS name.

> Demangling means DDS type name -> ROS type name?

Demangling means: DDS name -> ROS name.

----

- Functions that take an argument(s) which is a (topic or type) name:
  When the extra argument `no_mangle` is false, the passed name is a ros name that should be mangled into the DDS (or middleware specific) name. If `no_mangle` is true, the passed name is used directly as the middleware name.

- Function that as part of the output have a (topic or type) name:
  When the extra argument `no_demangle` is false, the output names will be demangled from DDS (or middleware specific) name, to the correct ros name. When it's true, that transformation is not done, and the raw name appears in the output.

Currently, we only had functions of the second type, that take an extra boolean called `no_demangle`[1]. This new functions take a name, and have names as part of their output. The idea is to have a boolean that means `no mangle inputs nor demangle outputs`. I originally suggested calling it `no_mangle`, as I find that more intuitive when a name is part of the input. But, it's really not intuitive. Besides the name, a big comment in the doc string is needed.

[1] That argument is incorrectly documented in `rmw` as `demangle`, though all the rmw implementations rename that argument as `no_demangle` (and they use that last convention). I will send a PR correcting that later.

Reply:
Sounds good.

I have inverted the logic here for demangling.